network-api/networkapi/wagtailpages/pagemodels/products.py

```diff
@@ -426,6 +480,86 @@ def get_context(self, request, *args, **kwargs):
         )} - {self.title}'''
         return context
 
+    def serve(self, request, *args, **kwargs):
+        # In Wagtail we use the serve() method to detect POST submissions.
+        # Alternatively, this could be a routable view.
+        if request.body and request.method == "POST":
+            # If the request is POST. Parse the body.
+            data = json.loads(request.body)
+            # If the POST body has a productID and value, it's someone voting on the product
+            if data['productID'] and data["value"]:
```

Comment:
accessing a non-existent dict key with throw a `KeyError`. Use `get` instead:
```
if data.get('productID') and data.get("value"):
```
Although as we're then also using those values inside the conditional body, let's cache them:

```suggestion
            product_id = data.get('productId', False)
            value = data.get('value', False)
            
            if product_id and value:
                try:
                    product_id = int(product_id)
                    value = int(value)
                except ValueError:
                    # generate an HttpResponseNotAllowed response and exit the function
```